### ISSUE TITLE
The loss of discriminator keeps decreasing

### ISSUE THREAD

**Issue by zhen8838** [TachibanaYoshino/AnimeGAN#17]
Hello, Can you help me?

I rewrite AnimeGAN by tensorflow2.0.  All hyperparameters are the same as your code.

But I found that in the training process, at first the generator generated the correct image, but later the loss of the discriminator gradually decreased, resulting in the loss of anime style of the generated image.

### loss value
![animegan-1](https://user-images.githubusercontent.com/26156999/79304409-d7d70f80-7f23-11ea-8865-efee4c6c7537.png)
### generate image 
![anime](https://user-images.githubusercontent.com/26156999/79304433-e6252b80-7f23-11ea-97c1-7808638e9583.gif)

### my opinions

At first, I thought the loss method was wrong, and then I tried Dragan, but the effect was not as good as lsgan.

I'm currently trying to increase `training rate` : Generator : Discriminator = 3 : 1.

### I'd appreciate it if you could give me some tips

**Comment by TachibanaYoshino** [comment]
All right，  My suggestion is that fake_loss in ` loss = real_loss + fake_loss + real_blur_loss * 0.1 + gray_loss`  should be appropriately weighted at the later period, such as 2.0, or 1.5~10.0. Because the two gray-scale adversarial losses in the above formula are calculated by gray-scale pictures containing animation style, the discriminator is easy to judge that the generated pictures containing style are fakes at the later period, thus making them more real.   
The new Hayao style and  Shinkai style datasets，as well as  the updated animegan will be released successively.

**Comment by zhen8838** [comment]
Thank you very much for your advice. I'll do the experiment next. 

At the same time, we are looking forward to your new works!

**Comment by zhen8838** [comment]
Hi..
I do some experiment , It is found that the discriminator not will be weaker when `fake_loss` is weighted, but the generated image will be more realistic. Then I tried to increase the weight of `style loss`

### exp1 : `style loss` weight = 30 , `fake_loss` weight = 1
![anime-3](https://user-images.githubusercontent.com/26156999/79732171-e5c2d100-8325-11ea-8cf0-f8f6c267ec12.gif)

### exp2 : `style loss` weight = 30 , `fake_loss` weight = 5
![anime-5](https://user-images.githubusercontent.com/26156999/79732208-f115fc80-8325-11ea-9b0f-4bbf9daa4076.gif)

### exp3: load exp1 weights,  `style loss` weight = 20 , `fake_loss` weight = 1
![anime-6](https://user-images.githubusercontent.com/26156999/79732873-dc863400-8326-11ea-83aa-f129775346b8.gif)

### exp1 and exp2 loss
![anime-10](https://user-images.githubusercontent.com/26156999/79733346-84036680-8327-11ea-8e73-8d01cb1f2dee.jpg)

### my opinions

I think may be need reduce `fake loss` weight... I'll keep trying.
Have you ever tried to add a dropout layer to the discriminator? Will it work?

**Comment by TachibanaYoshino** [comment]
I have not used dropout layer.  When you weight the fake loss, the g_adv_weight and d_adv_weight should also be 300.0.

**Comment by TachibanaYoshino** [comment]
Like the edge_blur adv loss is weighted by 0.1. My suggestion is to multiply the fake loss by a small weight such as 1.5 or 2.0 at the late training period, such as after 60 epoch, and keep all other weights unchanged. If the weight of style loss is set too large, it is easy to over stylize and lose content information. For example, the large text on the billboard becomes unclear and unrecognizable.  If you give the style loss a large weight such as 20~30, you may need  to train more epochs to prevent it from becoming over stylized.  According to past actual experience, I do not recommend too many epochs for style transfer training, and the continuous convergence of loss does not necessarily mean that the model's effect is getting better and better. I hope to find a balance between excessive stylization and reality.

**Comment by zhen8838** [comment]
OK. Thanks. Let me continue the experiment~

**Comment by zhen8838** [comment]
Due to the lack of more GPU and time, I will open the current results [here](https://github.com/zhen8838/AnimeGAN). 

At present, preliminary results have been achieved. The code supports multi GPU parallelism and mixed precision. If there is an opportunity, I hope to do research with you.
